packages/rest/src/controllers/credentials/CredentialController.ts

```diff
@@ -0,0 +1,167 @@
+import type { CredentialRecord } from "@aries-framework/core";
+
+import { Agent, CredentialPreview, JsonTransformer, RecordNotFoundError } from "@aries-framework/core";
+import {
+  Get,
+  Post,
+  JsonController,
+  Body,
+  InternalServerError,
+  Param,
+  NotFoundError,
+  Delete,
+} from "routing-controllers";
+import { Service, Inject } from "typedi";
+
+import { AcceptCredentialProposalRequest } from "../../schemas/AcceptCredentialProposalRequest";
+import { CredentialOfferRequest } from "../../schemas/CredentialOfferRequest";
+import { CredentialProposalRequest } from "../../schemas/CredentialProposalRequest";
+import { CredentialUtils } from "../../utils/CredentialUtils";
+
+@JsonController("/credentials")
+@Service()
+export class CredentialController {
+  @Inject()
+  private agent: Agent;
+  private credentialUtils: CredentialUtils;
+
+  public constructor(agent: Agent) {
+    this.agent = agent;
+    this.credentialUtils = new CredentialUtils(agent);
+  }
+
+  /**
+   * Retrieve credential record by credentialId
+   */
+  @Get("/:credentialId")
+  public async getCredentialById(@Param("credentialId") credentialId: string): Promise<CredentialRecord> {
+    try {
+      return await new CredentialUtils(this.agent).getCredentialById(credentialId);
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        throw new NotFoundError(`credential with credentialId "${credentialId}" not found.`);
+      }
+      throw new InternalServerError(`something went wrong: ${error}`);
+    }
+  }
+
+  /**
+   * Retrieve all credential records
+   */
+  @Get("/")
+  public async getAllCredentials() {
+    const credentials = await this.credentialUtils.getAllCredentials();
```

Comment:
Just one line return it.
```suggestion
    return await this.credentialUtils.getAllCredentials();
```

Reply:
```suggestion
    const credentials = await agent.credentials.getAll()
```